dlc/src/main/scala/org/bitcoins/dlc/BinaryOutcomeDLCClient.scala

```diff
@@ -0,0 +1,1056 @@
+
+  def cetExtPubKey(rootKey: ExtPublicKey, eventIndex: Int): ExtPublicKey = {
+    rootKey
+      .deriveChildPubKey(
+        BIP32Path(BIP32Node(1, hardened = false),
+                  BIP32Node(eventIndex, hardened = false)))
+      .get
+  }
+
+  val cetRemoteRefundPubKey: ECPublicKey =
+    cetExtPubKey(remoteExtPubKey, eventIndex = 0).key
+
+  val cetRemoteWinPubKey: ExtPublicKey =
+    cetExtPubKey(remoteExtPubKey, winIndex)
+
+  val cetRemoteLosePubKey: ExtPublicKey =
+    cetExtPubKey(remoteExtPubKey, loseIndex)
+
+  /** Total collateral amount */
+  private val totalInput = input + remoteInput
+
+  private val totalFunding =
+    fundingUtxos.foldLeft(0L)(_ + _.amount.satoshis.toLong)
+  private val remoteTotalFunding =
+    remoteFundingInputs.foldLeft(0L) {
+      case (accum, (_, output, _)) =>
+        accum + output.value.satoshis.toLong
+    }
+
+  /** Remote's payout in the Win case (in which Remote loses) */
+  val remoteWinPayout: CurrencyUnit = totalInput - winPayout
+
+  /** Remote's payout in the Lose case (in which Remote wins) */
+  val remoteLosePayout: CurrencyUnit = totalInput - losePayout
+
+  /** This is only used as a placeholder and we use an invariant
+    * when signing to ensure that this is never used.
+    *
+    * In the future, allowing this behavior should be done in TxBuilder.
+    */
+  private val emptyChangeSPK: ScriptPubKey = EmptyScriptPubKey
+
+  val fundingPubKey: ECPublicKey = fundingPrivKey.publicKey
+
+  val fundingSPK: MultiSignatureScriptPubKey = {
+    val fundingKeys = if (isInitiator) {
+      Vector(fundingPubKey, fundingRemotePubKey)
+    } else {
+      Vector(fundingRemotePubKey, fundingPubKey)
+    }
+
+    MultiSignatureScriptPubKey(2, fundingKeys)
+  }
+
+  /** Experimental approx. vbytes for a CET */
+  private val approxCETVBytes = 190
+
+  /** Experimental approx. vbytes for a closing tx spending ToLocalOutput */
+  private val approxToLocalClosingVBytes = 122
+
+  private val cetFee: CurrencyUnit = Satoshis(approxCETVBytes * feeRate.toLong)
+  private val toLocalClosingFee: CurrencyUnit = Satoshis(
+    approxToLocalClosingVBytes * feeRate.toLong)
+
+  private val isRBFEnabled = Policy.isRBFEnabled
+  private val sequence =
+    if (isRBFEnabled) UInt32.zero else TransactionConstants.sequence
+
+  lazy val createUnsignedFundingTransaction: Transaction = {
+    /* We need to commit to the CET's and local closing tx's fee during the construction of
+     * the funding transaction so that the CET outputs have the expected payouts.
+     *
+     * Once computed, we add the estimated amount to the fundingOutput so it can be used for fees later.
+     */
+    val halfCetFee = Satoshis((cetFee + toLocalClosingFee).satoshis.toLong / 2)
+
+    val output: TransactionOutput =
+      TransactionOutput(totalInput + halfCetFee + halfCetFee,
+                        P2WSHWitnessSPKV0(fundingSPK))
+
+    val (initiatorChange, initiatorChangeSPK, otherChange, otherChangeSPK) =
```

Comment:
important change: order of outputs now matters as both parties must generate the same txs